apache/superset
# fix: Save properties after applying changes in Dashboard

### SUMMARY
This PR fixes a severe issue for which changes in the properties modal of a Dashboard won't be fully saved when the `onlyApply` prop was used. This PR https://github.com/apache/superset/pull/17392 enabled the `onlyApply` tag. Later, manual tests have shown that the tag wasn't ready to be used as both the frontend and the backend were not fully implemented for that to work properly. In addition to that, this PR does the following:

- Refactors the `PropertiesModal` to use typescript
- Refactors the `form` component to use the functionalities provided by Antdesign
- Fixes an issue in the Dashboard detail page for which when applying changes and re-opening the properties modal it did not show the latest changes 
- Fixes an issue for which changes in the `json_metadata` for `color_namespace`, `expanded_slices`, `timed_refresh_immune_slices` were not saved 
- Fixes an issue for which the dashboard wasn't redirecting back to the original URL when the _slug_ was deleted
- Fixes an issue for which changes to native filters would not update the json_metadata immediately, causing potential overwrites
- Changes the endpoint that was used for updating the dashboard (`save_dash`) with the standard PUT endpoint. Also, it adds the `set_dash_metadata` function to the `update` command in the backend. The `set_dash_metadata` function was only used by the `save_dash` endpoint before, causing potential discrepancies in the way the `json_metatada` was handled by the standard PUT endpoint.
- Removes unnecessary fetch requests

### TESTING INSTRUCTIONS
1. Open a Dashboard
2. Edit the properties, including the json metadata
3. Apply the changes
4. Reopen the modal and make sure the latest changes are there
4. Finally, save the changes
5. Reload the Dashboard and make sure the properties were fully saved

### ADDITIONAL INFORMATION
